src/search-js/decorator.ts

```diff
@@ -0,0 +1,49 @@
+import { SearchProduct, SearchProductSku } from "@nosto/nosto-js/client"
+import { CurrencyConfig, getCurrencyFormatting } from "./currencies"
+
+type Prices = Pick<SearchProduct, "price" | "listPrice">
+
+type FormattedPrices = {
+  priceText?: string
+  listPriceText?: string
+}
+
+/**
+ * Exposes currency formatting logic as a SearchProduct decorator
+ */
+export function priceDecorator(config?: Partial<CurrencyConfig>) {
+  const { formatCurrency } = getCurrencyFormatting(config)
+
+  function formatPrices<T extends Prices>(obj: T, currency?: string) {
+    const formatted: FormattedPrices = {}
+    if (obj.price !== undefined) {
```

Comment:
I don't remember the type of `price` property. Can this be null?

Reply:
yes, if the price is not mapped in the request
all SerchProduct fields are optional